src/utils/refreshDockerode.ts

```diff
@@ -22,53 +25,122 @@ const SSH_URL_REGEX = /ssh:\/\//i;
+    const options: Dockerode.DockerOptions = {};
     let dockerContext: IDockerContext;
 
-    try {
-        ({ Context: dockerContext } = await dockerContextManager.getCurrentContext());
+    ({ DurationMs: actionContext.telemetry.measurements.contextRetrievalTimeMs, Result: { Context: dockerContext } } = await timeUtils.timeIt(async () => dockerContextManager.getCurrentContext()));
 
-        if (!newEnv.DOCKER_HOST) {
-            newEnv.DOCKER_HOST = dockerContext?.Endpoints.docker.Host;
-        }
+    if (dockerContext === undefined) { // Undefined context means there's only the default context
+        actionContext.telemetry.properties.hostSource = 'defaultContextOnly';
+    } else if (/^default$/i.test(dockerContext.Name)) {
+        actionContext.telemetry.properties.hostSource = 'defaultContextSelected';
+    } else {
+        actionContext.telemetry.properties.hostSource = 'customContextSelected';
+    }
 
-        if (!newEnv.DOCKER_TLS_VERIFY && dockerContext?.Endpoints.docker.SkipTLSVerify) {
-            // https://docs.docker.com/compose/reference/envvars/#docker_tls_verify
-            newEnv.DOCKER_TLS_VERIFY = "";
-        }
-    } catch (error) {
-        /* eslint-disable-next-line @typescript-eslint/no-floating-promises */
-        ext.ui.showWarningMessage(localize('vscode-docker.utils.dockerode.dockerContextUnobtainable', 'Docker context could not be retrieved.') + ' ' + parseError(error).message);
+    const host = dockerContext?.Endpoints?.docker?.Host;
+
+    if (host) {
+        const parsed = new url.URL(host);
+
+        options.host = host; // Intentionally the full URL (docker-modem can figure out the protocol and hostname from it)
+        options.port = parsed.port; // docker-modem can figure out the port if it is not explicit in the URL
+        options.username = parsed.username;
+
+        actionContext.telemetry.properties.hostProtocol = parsed.protocol;
+    } else {
+        // If the context doesn't have a Docker host, Dockerode will assume the default npipe://... or unix://...
+        actionContext.telemetry.properties.hostProtocol = os.platform() === 'win32' ? 'npipe:' : 'unix:';
     }
 
-    if (newEnv.DOCKER_HOST && SSH_URL_REGEX.test(newEnv.DOCKER_HOST)) {
-        // If DOCKER_HOST is an SSH URL, we need to configure / validate SSH_AUTH_SOCK for Dockerode
-        // Other than that, we use default settings, so return undefined
-        if (!await validateSshAuthSock(newEnv)) {
-            // Don't wait
-            /* eslint-disable-next-line @typescript-eslint/no-floating-promises */
-            ext.ui.showWarningMessage(localize('vscode-docker.utils.dockerode.sshAgent', 'In order to use an SSH DOCKER_HOST, you must configure an ssh-agent.'), { learnMoreLink: 'https://aka.ms/AA7assy' });
-        }
+    // Currently the environment variable is the only way to configure this in docker-modem
+    if (dockerContext?.Endpoints?.docker?.SkipTLSVerify) {
+        newEnv.DOCKER_TLS_VERIFY = '';

```

Comment:
Please retain the comment about the fact that to disable TLS (certificate) verification you actually need to **set it to empty value**.